### PR TITLE
Support psycopg 3.x

### PR DESCRIPTION
**Pull request by albireox**
Supports using `psycopg` 3.x (the [psycopg package](https://www.psycopg.org/psycopg3/docs/)) via the `peewee` `playhouse.psycopg3_ext` extension.

Although not implemented in `peewee`, `psycopg` 3 supports asynchronous connections out of the box, which may be useful to implement in some form.

This branch depends on #266 which should be merged first.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -30,14 +30,15 @@ content-type = "text/markdown"
 
 [project.optional-dependencies]
 all = [
+    "psycopg[binary]",
```
Comment (havok2063) [inline pyproject.toml:33]: Should we retain support for psycopg <3?   I think you can install 2.9 and 3+ at the same time.  Not sure if we need to, but I want to make sure we don't break any existing usage of `sdssdb`.

File: python/sdssdb/__init__.py [hunk 2]
```diff
@@ -32,6 +33,8 @@
 config = get_config(NAME)
 
 autoconnect = True
+use_psycopg3 = os.environ.get("SDSSDB_PSYCOPG3", "false")
```
Comment (havok2063) [inline python/sdssdb/__init__.py:36]: We may want to document this somewhere, what this flag is, and what it's needed for.

**Review by havok2063 — COMMENTED** [review]
Do we need to consider any changes to the `sqlalchemy` side of things for psycopg 3+ usage?

File: pyproject.toml (hunk 1 above)
Comment (albireox) [reply pyproject.toml:33]: Good point, I had not thought of that. According to [this](https://docs.sqlalchemy.org/en/20/dialects/postgresql.html#module-sqlalchemy.dialects.postgresql.psycopg) SQLA 2.x has native support so maybe that's something we can ensure we test when we update sdssdb to support it.

File: python/sdssdb/__init__.py (hunk 2 above)
Comment (albireox) [reply python/sdssdb/__init__.py:36]: They should. I'll add something to the docs.

**Review by havok2063 — APPROVED** [review]

**Pull request merged**